sycl/include/CL/sycl/detail/config.def

```diff
@@ -0,0 +1,14 @@
+// Singnature:
+// CONFIG(CONFIG_NAME, VALUE_MAX_SIZE_IN_BYTES, COMPILE_TIME_CONFIG_NAME)
```

Comment:
Nit: I would call it CONTROL_NAME instead of CONFIG_NAME to avoid ambiguity with SYCL_CONFIG_NAME (even though they're in different places).

Reply:
What if I rename SYCL_CONFIG_NAME to SYCL_CONFIG_FILE_NAME? Will it solve ambiguity?

Reply:
+1 to SYCL_CONFIG_FILE_NAME and leave CONFIG_NAME as is.